sdks/java/io/rabbitmq/src/main/java/org/apache/beam/sdk/io/rabbitmq/RabbitMqIO.java

```diff
@@ -233,26 +237,73 @@ public Read withQueue(String queue) {
 
     /**
      * You can "force" the declaration of a queue on the RabbitMQ broker. Exchanges and queues are
-     * the high-level building blocks of AMQP. These must be "declared" before they can be used.
-     * Declaring either type of object simply ensures that one of that name exists, creating it if
-     * necessary.
+     * the high-level building blocks of AMQP. These must be "declared" (created) before they can be
+     * used. Declaring either type of object ensures that one of that name and of the specified
+     * properties exists, creating it if necessary.
      *
-     * @param queueDeclare If {@code true}, {@link RabbitMqIO} will declare the queue. If another
-     *     application declare the queue, it's not required.
+     * <p>NOTE: When declaring a queue or exchange that already exists, the properties specified in
+     * the declaration must match those of the existing queue or exchange. That is, if you declare a
+     * queue to be non-durable but a durable queue already exists with the same name, the
+     * declaration will fail. When declaring a queue, RabbitMqIO will declare it to be non-durable.
+     *
+     * @param queueDeclare If {@code true}, {@link RabbitMqIO} will declare a non-durable queue. If
+     *     another application created the queue, this is not required and should be set to {@code
+     *     false}
      */
     public Read withQueueDeclare(boolean queueDeclare) {
       return builder().setQueueDeclare(queueDeclare).build();
     }
 
     /**
-     * Instead of consuming messages on a specific queue, you can consume message from a given
-     * exchange. Then you specify the exchange name, type and optionally routing key where you want
-     * to consume messages.
+     * In AMQP, messages are published to an exchange and routed to queues based on the exchange
+     * type and a queue binding. Most exchange types utilize the routingKey to determine which
+     * queues to deliver messages to. It is incumbent upon the developer to understand the paradigm
+     * in place to determine whether to declare a queue, with the appropriate binding should be, and
```

Comment:
```suggestion
     * in place to determine whether to declare a queue, what the appropriate binding should be, and
```

Reply:
(I'm not sure, this _could_ have been intended to be `[...]whether to declare a queue--with the appropriate binding--and[...]`)